kong/plugins/basic-rate-limiting/handler.lua

```diff
@@ -0,0 +1,23 @@
+-- Copyright (C) Kong Inc.
+local kong_meta = require "kong.meta"
+
+local kong = kong
+local ngx = ngx
+local shm = ngx.shared.kong_rate_limiting_counters
+
+local BasicRateLimitingHandler = {}
+
+BasicRateLimitingHandler.VERSION = kong_meta.version
+BasicRateLimitingHandler.PRIORITY = 911
+
+function BasicRateLimitingHandler:access(conf)
+    local counter = shm:incr("BasicRateLimiting",1,0,60)
```

Comment:
In the original rate limiting plugin, the key is based on the current timestamp, so every window aligns with the physical time. In the current implementation, they share the same key, so a window starts when the first request comes in. Not a big difference, I think the current way maybe even better.

- It makes the code simpler. Only one key is needed for every window size.
- The limit is more strict. For the original implementation, the rate could be doubled at the border of two windows.
- It makes the tests less flaky as they don't rely on the start time of test any more.